src/java.base/share/classes/sun/nio/fs/PollingWatchService.java

```diff
@@ -391,7 +389,7 @@ synchronized void poll() {
                     }
 
                     // check if file has changed
-                    if (e.lastModified != lastModified) {
+                    if (!e.lastModified().equals(lastModified)) {
```

Comment:
Hello Alan, with the change in this PR, the `lastModified` field in `CacheEntry` can now be null (because of any `IOException` on 362). This could then lead to a `NullPointerException` here, so I think we would need a null check here.